crates/ruma-events-macros/src/event_enum.rs

```diff
@@ -198,9 +200,38 @@ fn expand_any_with_deser(
         #event_deserialize_impl
 
         #redacted_enum
+
+        #from_impl
     })
 }
 
+fn expand_from_impl(
+    ident: Ident,
+    content: &[TokenStream],
+    variants: &[EventEnumVariant],
+) -> TokenStream {
+    let from_impls = content
+        .iter()
+        .zip(variants)
+        .map(|(content, variant)| {
+            let variant_tokens = variant.ident.to_token_stream();
+            let variant_attrs = &variant.attrs;
```

Comment:
Could be shortened from `variant_attrs` to just `attrs`, esp. if above is also renamed to `ident`.